src/base/worker_pool.h

```diff
@@ -10,18 +10,27 @@ namespace base {
 
 class WorkerPool {
  public:
-  using NetWorker = Fn<void(const Atomic<bool>&, Data&)>;
-  using SimpleWorker = Fn<void(const Atomic<bool>&)>;
+  using NetWorker = Fn<void(WorkerPool*, Data&)>;
```

Comment:
That's fine: make both mutex and cond. var mutable - they're not part of interface, are they?